distsql/distsql.go

```diff
@@ -101,23 +98,21 @@ func (r *selectResult) fetch(ctx goctx.Context) {
 		queryHistgram.WithLabelValues(label).Observe(duration.Seconds())
 	}()
 	for {
-		reader, err := r.resp.Next()
+		resultSubset, err := r.resp.Next()
 		if err != nil {
 			r.results <- resultWithErr{err: errors.Trace(err)}
 			return
 		}
-		if reader == nil {
+		if resultSubset == nil {
 			return
 		}
 		pr := &partialResult{
 			index:      r.index,
 			fields:     r.fields,
-			reader:     reader,
 			aggregate:  r.aggregate,
 			ignoreData: r.ignoreData,
-			done:       make(chan error, 1),
 		}
-		go pr.fetch()
```

Comment:
No, the old code assumes the operation is time-consuming, but it's not actually.